libs/helm/autocomplete/src/lib/hlm-autocomplete.ts

```diff
@@ -0,0 +1,244 @@
+import { BooleanInput } from '@angular/cdk/coercion';
+import {
+	booleanAttribute,
+	ChangeDetectionStrategy,
+	Component,
+	computed,
+	ElementRef,
+	forwardRef,
+	inject,
+	input,
+	linkedSignal,
+	output,
+	viewChild,
+} from '@angular/core';
+import { ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms';
+import { NgIcon, provideIcons } from '@ng-icons/core';
+import { lucideChevronDown, lucideSearch } from '@ng-icons/lucide';
+import { BrnCommandEmpty, CommandFilter } from '@spartan-ng/brain/command';
+import { hlm } from '@spartan-ng/brain/core';
+import { ChangeFn, TouchFn } from '@spartan-ng/brain/forms';
+import { BrnPopover, BrnPopoverContent } from '@spartan-ng/brain/popover';
+import { HlmCommandEmpty, HlmCommandGroup, HlmCommandItem, HlmCommandList } from '@spartan-ng/helm/command';
+import { HlmIcon } from '@spartan-ng/helm/icon';
+import { HlmPopoverContent } from '@spartan-ng/helm/popover';
+import type { ClassValue } from 'clsx';
+import { HlmAutocompleteCommand } from './hlm-autocomplete-command';
+import { HlmAutocompleteSearch } from './hlm-autocomplete-search';
+import { HlmAutocompleteSearchInput } from './hlm-autocomplete-search-input';
+import { HlmAutocompleteTrigger } from './hlm-autocomplete-trigger';
+
+export const HLM_AUTOCOMPLETE_VALUE_ACCESSOR = {
+	provide: NG_VALUE_ACCESSOR,
+	useExisting: forwardRef(() => HlmAutocomplete),
+	multi: true,
+};
+
+let nextId = 0;
+
+@Component({
+	selector: 'hlm-autocomplete',
+	imports: [
+		BrnPopover,
+		BrnPopoverContent,
+		HlmPopoverContent,
+
+		HlmCommandList,
+		HlmCommandGroup,
+		HlmCommandItem,
+		BrnCommandEmpty,
+		HlmCommandEmpty,
+
+		HlmAutocompleteCommand,
+		HlmAutocompleteSearch,
+		HlmAutocompleteSearchInput,
+		HlmAutocompleteTrigger,
+
+		NgIcon,
+		HlmIcon,
+	],
+	providers: [HLM_AUTOCOMPLETE_VALUE_ACCESSOR, provideIcons({ lucideSearch, lucideChevronDown })],
+	template: `
+		<brn-popover
+			align="start"
+			autoFocus="first-heading"
+			sideOffset="5"
+			closeDelay="100"
+			[closeOnOutsidePointerEvents]="true"
+		>
+			<hlm-autocomplete-command [filter]="filter()">
```

Comment:
@ashley-hunter this filter is used to set the items to visible/invisible. Should we just set it to true = visible, because options are filtered already?

Reply:
Yeah, I think that makes sense